### PR TITLE
Update search profile API documentation with concurrency related changes

### LINKED ISSUE #4281: [DOC] Update search profile API documentation with concurrency related changes
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 As a part of introducing concurrent segment search we have also updated the search profile API to return more detailed summary stats in cases with concurrency.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/OpenSearch/issues/7355
https://github.com/opensearch-project/OpenSearch/pull/7898

### PR DESCRIPTION
**Pull request by ticheng-aws**
### Description
We need to update the document for the query profile part [here](https://opensearch.org/docs/latest/api-reference/profile#the-query-array-1) in the next release. We will have some new fields similar to the [aggregation profile](https://opensearch.org/docs/latest/api-reference/profile#the-aggregations-array) like `max/min/avg_slice_time_in_nanos` and time/count stats for the query breakdowns except `create_weight` type. More details can be found [#9248](https://github.com/opensearch-project/OpenSearch/pull/9248).

### Issues Resolved
Resolves #4281 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @ticheng-aws! Please confirm the technical accuracy of my changes, and then we'll be ready to move the PR to editorial review.

File: _api-reference/profile.md [hunk 1]
```diff
@@ -987,18 +987,18 @@ The following sections contain definitions of all modified or added response fie
...
-|`breakdown.<method>_count`	|For concurrent segment search, this field contains the total number of invocations of a `<method>` obtained by adding the number of method invocations for all segments.       |
-|`breakdown.max_<method>_count`	|The maximum number of invocations of a `<method>` on any slice. `create_weight` method does not have profiled avg count stats because they are executed at the query-level rather than the slice-level.       |
-|`breakdown.min_<method>_count`	|The minimum number of invocations of a `<method>` on any slice. `create_weight` method does not have profiled avg count stats because they are executed at the query-level rather than the slice-level.       |
-|`breakdown.avg_<method>_count`	|The average number of invocations of a `<method>` on any slice. `create_weight` method does not have profiled avg count stats because they are executed at the query-level rather than the slice-level.       |
+|`time_in_nanos`	| The total elapsed time for this query, in nanoseconds. For concurrent segment search, `time_in_nanos` is the total amount of time across all slices (`max(slice_end_time) - min(slice_start_time)`).   |
```
Comment (sohami) [inline _api-reference/profile.md:990]: `slice_end_time` and `slice_start_time` is not exposed/known to users. We should instead replace it with wording something like: _For concurrent segment search, `time_in_nanos` is the total time spent across all the slices i.e. difference between end time of last completed slice execution and start time of first slice execution._

File: _api-reference/profile.md [hunk 2]
```diff
@@ -987,18 +987,18 @@ The following sections contain definitions of all modified or added response fie
...
+|`max_slice_time_in_nanos`	| The maximum amount of time taken by any slice to run a query, in nanoseconds.	|
+|`min_slice_time_in_nanos`	| The minimum amount of time taken by any slice to run a query, in nanoseconds.	|
+|`avg_slice_time_in_nanos`	| The average amount of time taken by any slice to run a query, in nanoseconds.	|
+|`breakdown.<method>`	| For concurrent segment search, this is the total elapsed time across all slices (`max(slice_end_time) - min(slice_start_time)`). For example, for the `build_scorer` method, it is the total time spent constructing the `Scorer` object across all slices. |
```
Comment (sohami) [inline _api-reference/profile.md:994]: Same as above

File: _api-reference/profile.md [hunk 3]
```diff
@@ -936,9 +987,18 @@ The following sections contain definitions of all modified or added response fie
...
+|`breakdown.max_<method>`	| The maximum amount of time taken by any slice to run a query method. Breakdown stats for the `create_weight` method do not include profiled `max` time because the method runs at the query level rather than the slice level.	|
+|`breakdown.min_<method>`	| The minimum amount of time taken by any slice to run a query method. Breakdown stats for the `create_weight` method do not include profiled `min` time because the method runs at the query level rather than the slice level.  |
+|`breakdown.avg_<method>`	| The average amount of time taken by any slice to run a query method. Breakdown stats for the `create_weight` method do not include profiled `avg` time because the method runs at the query level rather than the slice level.	|
+|`breakdown.<method>_count`	| For concurrent segment search, this field contains the total number of invocations of a `<method>` obtained by adding the number of method invocations for all segments.       |
```
Comment (sohami) [inline _api-reference/profile.md:998]: `obtained by adding the number of method invocations for all segments.` Is this segments of slices here ?

File: _api-reference/profile.md (hunk 1 above)
Comment (ticheng-aws) [reply _api-reference/profile.md]: Hi @sohami, thanks for the review. It's been addressed now.

File: _api-reference/profile.md (hunk 2 above)
Comment (ticheng-aws) [reply _api-reference/profile.md]: It's been addressed now.

File: _api-reference/profile.md (hunk 3 above)
Comment (ticheng-aws) [reply _api-reference/profile.md:998]: Hi @sohami, you are right.

Comment (sohami) [reply _api-reference/profile.md:998]: Sorry there was type in my previous comment. I wanted to confirm if this is segments **or** slices here ?

Comment (ticheng-aws) [reply _api-reference/profile.md]: Hi @sohami, I've updated to use "for all slices" instead. Thanks for the review.

**Review by natebower — APPROVED** [review]
@kolchfa-aws LGTM

**Review by sohami — APPROVED** [review]

**Pull request merged**